boards/FSM/Src/App/App_SetPeriodicCanSignals.c

```diff
@@ -143,7 +147,69 @@ void App_SetPeriodicSignals_AcceleratorPedal(const struct FsmWorld *world)
 
 void App_SetPeriodicSignals_MotorShutdownFaults(const struct FsmWorld *world)
 {
-    struct FsmCanTxInterface *can_tx = App_FsmWorld_GetCanTx(world);
+    struct FsmCanTxInterface *can_tx      = App_FsmWorld_GetCanTx(world);
+    struct FsmCanRxInterface *can_rx      = App_FsmWorld_GetCanRx(world);
+    static uint8_t            error_count = 0;
+
+    // Perform torque plausibility check if DCM is in the drive state
+    if (App_CanRx_DCM_STATE_MACHINE_GetSignal_STATE(can_rx) ==
+        CANMSGS_DCM_STATE_MACHINE_STATE_DRIVE_CHOICE)
+    {
+        const bool is_regen_allowed =
+            App_CanTx_GetPeriodicSignal_LEFT_WHEEL_SPEED(can_tx) >
+                REGEN_WHEEL_SPEED_THRESHOLD_KPH &&
+            App_CanTx_GetPeriodicSignal_RIGHT_WHEEL_SPEED(can_tx) >
+                REGEN_WHEEL_SPEED_THRESHOLD_KPH;
+        const float regen_paddle_percentage =
+            (float)App_CanRx_DIM_REGEN_PADDLE_GetSignal_MAPPED_PADDLE_POSITION(
+                can_rx);
+        float expected_torque_request;
+
+        if (regen_paddle_percentage > 0.0f && is_regen_allowed)
+        {
+            expected_torque_request =
+                -0.01f * regen_paddle_percentage * MAX_TORQUE_REQUEST_NM;
+        }
+        else
+        {
+            expected_torque_request =
+                0.01f *
+                App_CanTx_GetPeriodicSignal_MAPPED_PEDAL_PERCENTAGE(can_tx) *
+                MAX_TORQUE_REQUEST_NM;
+        }
```

Comment:
anywhere to share this between DCM and FSM?

Reply:
created a new module in the shared folder named `App_SharedTorqueRequest`